background.js

```diff
@@ -1,29 +1,62 @@
+// Function to get the current tab
+async function getCurrentTab() {
+    let queryOptions = { active: true, currentWindow: true };
+    let [tab] = await chrome.tabs.query(queryOptions);
+    return tab;
+}
 
-// Called when the user clicks on the browser action.
-chrome.browserAction.onClicked.addListener(function(tab) {
-    var link = '[' + tab.title + '](' + tab.url + ')';
+function injectedFunction(tab) {
+    title = tab.title;
+    chrome.storage.sync.get(null, (storage) => {
+        for (index in storage.fieldsets) {
+            regexUrl = new RegExp(storage.fieldsets[index].url, 'gi');
+            if (regexUrl.test(tab.url)) {
+                regexSearch = new RegExp(storage.fieldsets[index].search, 'gi');
+                regexReplace = new RegExp(storage.fieldsets[index].replace, 'i');
+                if (regexSearch.test(title)) {
+                    title = title.replace(regexSearch, storage.fieldsets[index].replace);
+                    // we don't stop in case there are several rules
+                }
+            }
+        }
 
-    element = document.createElement("textarea");
-    element.value = link;
+        var link = '[' + title + '](' + tab.url + ')';
 
-    // It's required to put the temporary element to document tree before copy command
-    // See: https://stackoverflow.com/questions/25622359/clipboard-copy-paste-on-content-script-chrome-extension
-    document.body.appendChild(element);
-    element.select();
-    document.execCommand('copy');
+        let input = document.createElement('textarea');
+        document.body.appendChild(input);
+        input.value = link;
+        input.focus();
+        input.select();
+        document.execCommand('copy');
+        input.remove();
+    });
+
+}
 
-    element.remove();
+// Listen for hotkey shortcut command
+chrome.action.onClicked.addListener(function (tab) {
 
-    chrome.browserAction.setBadgeText({
-        text: "Done",
-        tabId: tab.id
+    // Callback to wait for chrome to get the current tab and then pass the tab into the injection script for copying to clipboard
+    chrome.scripting.executeScript({
+        target: { tabId: tab.id },
+        func: injectedFunction,
+        args: [tab],
     });
 
-    var clear_callback = function () {
-        chrome.browserAction.setBadgeText({
-            text: "",
-            tabId: tab.id
+});
+
+// Listen for hotkey shortcut command
+chrome.commands.onCommand.addListener((getCurrentLink) => {
```

Comment:
`getCurrentLink` is the parameter for command ID and  is not checked, should validate it. :D

Also `commandId` should be a better name for it.